src/cache/cache.rs

```diff
@@ -47,6 +47,7 @@ fn get_file_mode(file: &fs::File) -> Result<Option<u32>> {
 }
 
 #[cfg(windows)]
+#[allow(clippy::unnecessary_wraps)]
```

Comment:
Smells like an `impl Trait` is needed around here, but that's unrelated to this patch.